solidity/contracts/isms/multisig/AbstractMultisigIsm.sol

```diff
@@ -122,4 +82,37 @@ abstract contract AbstractMultisigIsm is IMultisigIsm {
         }
         return true;
     }
+
+    function computeDigest(bytes calldata _metadata, bytes calldata _message)
+        internal
+        pure
+        returns (bytes32)
+    {
+        bytes32 signedRoot;
+        bytes32 signedMessageId;
+
+        if (
+            MultisigIsmMetadata.suffixType(_metadata) ==
+            MultisigIsmMetadata.SuffixType.ROOT
+        ) {
+            signedRoot = MultisigIsmMetadata.root(_metadata);
+            signedMessageId = Message.id(_message);
+        } else {
+            signedRoot = MerkleLib.branchRoot(
+                Message.id(_message),
+                MultisigIsmMetadata.merkleProof(_metadata),
+                Message.nonce(_message)
+            );
+            signedMessageId = MultisigIsmMetadata.signedMessageId(_metadata);
+        }
+
+        return
+            CheckpointLib.digest(
+                Message.origin(_message),
+                MultisigIsmMetadata.originMailbox(_metadata),
+                signedRoot,
+                MultisigIsmMetadata.index(_metadata),
```

Comment:
Ah I realized the index is also optional iff no proof is included, it's just Message.nonce(_message)